sklearn/gaussian_process/tests/test_kernels.py

```diff
@@ -46,6 +45,12 @@
     if metric in ["additive_chi2", "chi2"]:
         continue
     kernels.append(PairwiseKernel(gamma=1.0, metric=metric))
+kernels += [SelectDimensionKernel(RBF(length_scale=2.0),
+                                  active_dim=[0]),
+            SelectDimensionKernel(RBF(length_scale=[2.0, 1.0]),
+                                  active_dim=[0, 1]),
+            SelectDimensionKernel(Matern(length_scale=0.5, nu=0.5),
```

Comment:
Should we also test for compound kernels such as Product or Sum, which was what the issue was all about?